src/category_theory/lifting_properties.lean

```diff
@@ -0,0 +1,208 @@
+- `id_has_right_lifting_property`: any identity has the rlp
+- `right_lifting_property_initial_iff`: spells out the rlp against a map whose source is an 
+initial object
+- `right_lifting_subcat`: given a functor F : D ⥤ arrow C, we construct the subcategory of those 
+morphisms p in C that satisfy the rlp w.r.t. F(i), for any object i in D.
+
+## Tags
+lifting property
+-/
+
+namespace category_theory
+
+universes v u
+variables {C : Type u} [category.{v} C]
+variables {D : Type u} [category.{v} D]
+
+variables {X Y Z : C}
+
+/- The left lifting property of a morphism i vs. a morphism p. -/
+def has_lifting_property (i p : arrow C) : Prop := ∀ (sq : i ⟶ p), arrow.has_lift sq
+
+/- Any isomorphism has the right lifting property with respect to any map. 
+A    → X 
+↓i    ↓p≅
+B    → Y
+-/
+lemma iso_has_right_lifting_property (i : arrow C) (p : X ≅ Y) : 
+has_lifting_property i (arrow.mk p.hom) :=
+begin
+  intro sq,
+  fconstructor,
+  fconstructor,
+  fconstructor,
+  { use sq.right ≫ p.inv }, -- the lift is obtained by p⁻¹ ∘ (B → Y)
+  { rw [←category.assoc, (arrow.w sq).symm],
+    simp, },
+  { simp, }
+end
+
+/- Any identity has the right lifting property with respect to any map. -/
+lemma id_has_right_lifting_property (i : arrow C) : has_lifting_property i (arrow.mk (𝟙 X)) :=
+begin
+  haveI := is_iso.id X,
+  exact iso_has_right_lifting_property i (category_theory.as_iso (𝟙 X)),
+end
+
+/- An equivalent characterization for right lifting against a map i whose source is initial. 
+∅ → X
+↓   ↓
+B → Y has a lifting iff there is a map B → X making the right part commute.
+-/
+lemma right_lifting_property_initial_iff (i p : arrow C) 
+(h : category_theory.limits.is_initial i.left) :
+has_lifting_property i p ↔ ∀ {e : i.right ⟶ p.right}, ∃ l : i.right ⟶ p.left, l ≫ p.hom = e :=
+begin
+  split,
+  { intros hlift bottom,
+    have comm : (category_theory.limits.is_initial.to h p.left) ≫ p.hom = i.hom ≫ bottom := 
+    begin apply category_theory.limits.is_initial.hom_ext h, end,
+    haveI := hlift (arrow.hom_mk comm),
+    use (arrow.has_lift.struct (arrow.hom_mk comm)).lift,
+    rw (arrow.has_lift.struct (arrow.hom_mk comm)).fac_right,
+    refl },
+  { intros h1 sq,
+    cases h1 with e he,
+    fconstructor,
+    fconstructor,
+    fconstructor,
+    { exact e },
+    { apply category_theory.limits.is_initial.hom_ext, simp, exact h },
+    { exact he } }
+end
+
+/- A helper construction: given a square between i and f ≫ g, produce a square between i and g, 
+whose top leg uses f:
+A  → X  
+     ↓f
+↓i   Y             --> A → Y 
+     ↓g                ↓i  ↓g
+B  → Z                 B → Z 
+ -/
+def square_to_second {i : arrow C} {f : X ⟶ Y} {g : Y ⟶ Z} (sq : i ⟶ arrow.mk (f ≫ g)) :
+i ⟶ arrow.mk g :=
+{ left := sq.left ≫ f,
+  right := sq.right,
+  w' := begin
+    have : sq.left ≫ f ≫ g = i.hom ≫ sq.right := arrow.w sq,
+    rw ←category.assoc at this,
+    tidy,
+  end
+}
```

Comment:
```suggestion
  w' := by simpa using arrow.w sq, }
```
I'm annoyed I can't get `tidy` to just cope with this one.

Reply:
Ah! If you add
```
-- `w_mk_left` is not needed, as it is a consequence of `w` and `mk_hom`.
@[simp, reassoc] lemma w_mk_right {f : arrow T} {X Y : T} {g : X ⟶ Y} (sq : f ⟶ mk g) :
  sq.left ≫ g = f.hom ≫ sq.right :=
sq.w
```
immediately after `lemma w` in `category_theory.arrow`, you can then just omit the `w' := ` field in this structure.

Reply:
Done. (Now in arrow.lean)